### ISSUE TITLE
Proposal for beating WhatsApp and supporting open communication

### ISSUE THREAD

**Issue by arnowelzel** [redsolution/xabber-android#634]
Also see https://github.com/siacs/Conversations/issues/1802

This is not an issue in the existing version - more a proposal how we may get Jabber/XMPP to become a **real** alternative to WhatsApp, since I fear that people now think WhatsApp is "good" and "secure" as it introduced end-to-end-encryption. But with WhatsApp people are still depending on one single company - Facebook. And this should not continue like this. But just telling people, they should move to Jabber doesn't really help since many users find Jabber too complicated and only suitable for "tech-savvy geeks".

_What makes WhatsApp so popular?_

Well - you don't need to create any account using a username/password and you don't need to know the account names of your friends - their phone numbers are enough to check, if they use WhatsApp as well and to reach them. And as you already have the numbers in your phonebook you literally can start to chat with your friends instantly after installing WhatsApp.

_How to achieve the same goal with XMPP/Jabber?_

My idea:

1) Offer to enter an existing account OR to create an account using one of several well known Jabber providers (e.h. jabber.org, jabber.de etc.) with a password which is automatically created locally and which can be displayed or saved for later reference. The list of providers could be loaded dynamically from a server (see point 3) where providers could register their service.

2) Detect the local phone number and/or allow the user a way to enter and confirm it manually.

3) Provide an online service for a central registry where the client can send the hash the telephone number together with the Jabber-ID.

4) Offer a function where the use can look up existing (hashed) phone numbers at the central registry to see, if others in his addressbook already use the service and if yes which Jabber-ID they have - so you can easily get in touch with others without the need of manually exchanging Jabber-IDs.

I'm aware, that the central registry may be seen as "evil" - but this should be implemented as open source and should be run by a non profit organisation. To cover the cost for the servers, backup, maintenance etc. one could either rely on donations or ask for a yearly usage fee by the users with the first 1 to 3 months for free - if a user decides not to pay, the registered hash gets locked with a status "user does not want to use the service any longer", so others will not find it any longer. However - communication is still possible, as the central registry is only needed to find others more easily, not for the communication itself. Of course you still can contact Jabber-IDs directly without using a registry server.

It's only a very rough proposal and there may still be things I forgot - but what do you think? Does this sound interesting? Did anyone try this yet?

I've been a professional software developer for the last 20+ years (see [http://arnowelzel.de/wp/en/about-me](http://arnowelzel.de/wp/en/about-me)). I have experience in Android development as well as in running and maintaining Linux servers (I run my own Jabber server as well as a number of domains and services for friends and family members) - so I can help wherever needed. But I don't want to do this without the support of others.

**Comment by krushmp** [comment]
Branding.  Get people away from sms texting by calling this Jabber Texting.  

2
NO
NOPE

You want a third party service like GPG key server go wild.  Pairing my mobile number is very opposite privacy.  

Perhaps you want gtalk: bastardized xmpp with all the free privacy raping features you want.

**Comment by arnowelzel** [comment]
I did NOT ask to use the phone number without user confirmation. Of
course this has to be confirmed by the user.

If you don't want to use this way of identification - then don't use it.
But for many people it would make a messenger on mobile devices much
easier to use.

Why do you think WhatsApp got that big? There is no "texting" in the
name, no "SMS" - just the fact that you can immediately reach all your
friends without asking anybody for their messenger ID - this is what
made WhatsApp successful.

I don't want anything - I'm happy with my own Jabber server. I just
thought about ways how one could establish an OPEN alternative to WhatsApp.

**Comment by climf** [comment]
branding

Help people using legacy sms texting embrace jabber texting.  VoLTE is jabber and sip.  "You are using jabber now".  I invite you to explicitly use jabber texting

legacy sms texting is needlessly costsly

you pay for data. legacy sms texting is delivered via IMS.  Don't pay twice.

Without legacy sms texting your monthly bill could be reduced by 50%

**Comment by climf** [comment]
your 2, 3, and 4 are antithetical to privacy

but it does seem like you are trying to reinvent the e.164 wheel

**Comment by arnowelzel** [comment]
@climf: I supposed this to be **optional**. If one decides to give up some privacy to get easier in touch with others, it should of course always be the decision of the user, not the software. And yes, I learned about E.164 (also see [https://tools.ietf.org/html/rfc4979]) already and that nobody seems to implement such a thing.

But this does not solve the problem, that people just don't know how to find others easily. That's the major reason why WhatsApp got so successful - you just install it and you have immediate communication with all your friends and collegues as long as you have a mobile number of them in your local address book.

Most users don't want to deal with another user account just for sending text messages and they don't want to deal with the process to tell everybody their current JID and ask everybody to get in touch this way, when you already have a mobile number.

But I understand that XMPP/Jabber is not aimed to be a messenger for the masses and not to replace WhatsApp or similar Apps (Signal, Threema, Facebook Messenger et.c.). I also learned that "standard" is not important in this market - things don't become successful because of standards.
